src/modules/InstURL.rb

```diff
@@ -86,24 +87,27 @@ def RewriteCDUrl(url)
     def installInf2Url(extra_dir = "")
       return @installInf2Url unless @installInf2Url.nil?
 
-      parts = URL.Parse(Linuxrc.InstallInf("ZyppRepoURL"))
+      zypp_repo_url = Linuxrc.InstallInf("ZyppRepoURL").to_s
 
-      if parts.empty?
+      if zypp_repo_url.empty?
         # Make it compatible with the current behaviour when
         # install.inf does not exist.
         log.warn "No URL specified through ZyppRepoURL"
-        parts = URL.Parse("cd:///")
+        zypp_repo_url = "cd:/"
       end
 
-      parts["path"] = File.join(parts["path"], extra_dir) unless extra_dir.empty?
+      uri = URI(zypp_repo_url)
```

Comment:
Well, I'd still prefer not parsing the URL unless really needed. The original code simply returned the `Linuxrc.InstallInf("ZyppRepoURL")` value.

The building URL code following later is actually not used for some time, it's dead code. It was kept just for compatibility reasons (IIRC some openSUSE release switched to `ZyppRepoURL` but the paralel SLE was still building the parts. Now we use `ZyppRepoURL` everywhere, we can drop that URL build part.)